File: Problems/Find the Key of the Numbers.py
class Solution(object):
    def generateKey(self, num1, num2, num3):
        vec = [str(num1), str(num2), str(num3)]
        
        for i in range(3):
            vec[i] = vec[i].zfill(4)  
        
        ans = "" 
        
        for i in range(4):
            ch = min(vec[0][i], vec[1][i], vec[2][i])
            ans += ch

        return int(ans)

File: Problems/Find the Key of the Numbers.cpp
int generateKey(int num1, int num2, int num3) {
    int key = 0, factor = 1;

    for (int i = 0; i < 4; i++) {
        int d1 = num1 % 10;
        int d2 = num2 % 10;
        int d3 = num3 % 10;
        key += (factor * min({d1, d2, d3}));
        factor *= 10;

        num1 /= 10;
        num2 /= 10;
        num3 /= 10;
    }

    return key;
}
